setup.php

```diff
@@ -81,6 +84,45 @@ function plugin_init_uninstall() {
             }
          }
 
+         $PLUGIN_HOOKS['pre_item_update']['uninstall'] = [
+             'Config' => ['PluginUninstallConfig', 'preConfigSet']
+         ];
+          $PLUGIN_HOOKS['pre_item_add']['uninstall'] = [
+              'Config' => ['PluginUninstallConfig', 'preConfigSet']
+          ];
+
+         $PLUGIN_HOOKS[Hooks::STALE_AGENT_CONFIG]['uninstall'] = [
+             [
+                 'label' => 'Apply uninstall profile',
+                 'item_action' => true,
+                 'render_callback' => static function ($config) {
+                     $stale_agents_uninstall = Config::getConfigurationValue('plugin:uninstall', 'stale_agents_uninstall');
+                     if (!\PluginUninstallModel::canView()) {
+                         return false;
+                     }
+                     return \PluginUninstallModel::dropdown([
+                         'name' => '_stale_agents_uninstall',
+                         'value' => $stale_agents_uninstall ?? 0,
+                         'entity' => $_SESSION['glpiactive_entity'],
+                         'condition' => [
+                             'types_id' => \PluginUninstallModel::TYPE_MODEL_UNINSTALL
+                         ],
+                         'display' => false,
+                     ]);
+                 },
+                 'action_callback' => static function (?Agent $agent, array $config, ?CommonDBTM $item): bool {
+                     if ($item === null) {
+                         return false;
+                     }
+                     $stale_agents_uninstall = Config::getConfigurationValue('plugin:uninstall', 'stale_agents_uninstall');
+                     \PluginUninstallUninstall::uninstall(get_class($item), $stale_agents_uninstall, [
+                         get_class($item) => [$item->fields['id'] => true]
+                     ], '');
```

Comment:
Logic that alter the item in `PluginUninstallUninstall::uninstall()` should probably be moved into a new method. Indeed, in cron context, we should not generate any output (progress bar and operation summary).